lib/tests/streamlit/components_test.py

```diff
@@ -241,17 +246,70 @@ def test_json_and_df_args(self):
         self.assertIsNotNone(proto.args_dataframe)
 
     def test_key(self):
-        """Test the 'key' param"""
-        # TODO!
-        pass
+        """Test the 'key' param."""
+        self.test_component(foo="bar", key="baz")
+        proto = self.get_delta_from_queue().new_element.component_instance
+        self.assertEqual("baz", proto.id.split("-")[0])
 
     def test_default(self):
-        """Test the 'default' param"""
-        # TODO!
-        pass
+        """Test the 'default' param."""
+        component = self.test_component(foo="bar", default="baz")
```

Comment:
How about `return_value` instead of `component` for this variable name?

Reply:
Done!